### PR TITLE
Added in checking formatting and clippy

### PR DESCRIPTION
**Pull request by fhaynes**
Coverage will be done in another commit.

### CODE REVIEW

**Comment by jstnlef** [comment]
Seems ok 👍

File: Jenkinsfile [hunk 1]
```diff
@@ -1,6 +1,22 @@
...
+                CARGO_HOME = '/home/jenkins/.cargo'
+                RUSTUP_HOME = '/home/jenkins/.rustup'
+                RUSTFLAGS = "-D warnings"
+            }
+            agent {
+                label 'linux'
+            }
+            steps {
+                echo 'Checking formatting...'
+                sh '$CARGO_HOME/bin/cargo fmt -- --check'
```
Comment (LucioFranco) [inline Jenkinsfile]: is it possible to break these out into different steps? I find it easier to parse the issues then.

Comment (fhaynes) [reply Jenkinsfile]: Those are different steps. Each command is a step.

Comment (fhaynes) [reply Jenkinsfile]: Oh, if you mean stage, sure.

**Pull request merged**